add camera modes hint

odudex opened the pull request:
### What is this PR for?
Add camera mode toggle hint and refactor camera image rendering

### What is the purpose of this pull request?
- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [X] Other

tadeubas commented:
Awesome idea! Tested on Cube, Dock, Yahboom and Amigo here without any issues. Nice job! :+1:

jdlcdl commented:
Tried this on amigo and --unsupported yahboom_devkit, worked very well.  Also worked well on simulator for amigo.

Not sure what to expect for other devices, but simulator for yahboom overlaid the camera image on-top of the mode hints, which was not the case on the yahboom_devkit (while I expect that yahboom and yahboom_devkit have similar screens).

These hints are great to uncover very useful features (zoom mode relieves me from having to refocus camera on my amigo often).

odudex commented:
Simulator: Fixed simulator for Yahboom, Dock, also M5stickV which was not showing anything at all.
I also changed the color settings as camera feed was bluish on my PC, please let me know if colors got messed up for you.
